.github/workflows/main.yml

```diff
@@ -48,35 +48,53 @@ jobs:
           repository: "mcagov/beacons-webapp"
       - id: get_webapp_commit_hash
         name: Get commit hash from Webapp repo
+        if: github.event.inputs.updateImageTags == 'true'
         run: echo "::set-output name=webapp_commit_hash::$(git rev-parse HEAD)"
       - uses: actions/checkout@v2
         with:
           repository: "mcagov/beacons-service"
       - id: get_service_commit_hash
         name: Get commit hash from Service repo
+        if: github.event.inputs.updateImageTags == 'true'
         run: echo "::set-output name=service_commit_hash::$(git rev-parse HEAD)"
+      - uses: actions/checkout@v2
+      - uses: hashicorp/setup-terraform@v1
+      - name: Save image tags for dev environment
+        if: github.event.inputs.updateImageTags == 'true'
+        run: |
+          cd terraform
+          echo "webapp_image_tag = \"${{ steps.get_webapp_commit_hash.outputs.webapp_commit_hash }}\"" > dev.images.tfvars
+          echo "service_image_tag = \"${{ steps.get_service_commit_hash.outputs.service_commit_hash }}\"" >> dev.images.tfvars
+          terraform fmt
+          git config user.name github-actions
+          git config user.email github-actions@github.com
+          git add .
+          git commit -m "Update image tags used in dev environment [skip ci]"
+          git push
 
-  terraform_deploy:
-    needs: [terraform_lint, get_docker_image_tags, prettier_check]
+  terraform_deploy_dev:
+    needs: [terraform_lint, get_and_save_docker_image_tags, prettier_check]
     if: github.ref == 'refs/heads/main'
     name: Terraform Deploy
     runs-on: ubuntu-latest
+    environment: development
     defaults:
       run:
         working-directory: terraform
     env:
       AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
       AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      TF_WORKSPACE: dev
     steps:
       - uses: actions/checkout@v2
       - uses: hashicorp/setup-terraform@v1
       - name: Terraform Init
         run: terraform init
+      - name: Terraform format
+        run: terraform fmt
       - name: Terraform Validate
         run: terraform validate
       - name: Terraform Deploy
         env:
-          TF_VAR_service_image_tag: ${{ needs.get_docker_image_tags.outputs.service_commit_hash }}
-          TF_VAR_webapp_image_tag: ${{ needs.get_docker_image_tags.outputs.webapp_commit_hash }}
-          TF_VAR_db_password: ${{ secrets.DEV_DB_PASSWORD }}
-        run: terraform apply -auto-approve
+          TF_VAR_db_password: ${{ secrets.DB_PASSWORD }}
```

Comment:
Here we're using the password set in the `development` [environment in this repo](https://github.com/mcagov/beacons-integration/settings/environments/195657563/edit)!! 🎉